runtime/lua/lsp/autocmds.lua

```diff
@@ -0,0 +1,122 @@
+-- This file contains a table of all the functions that should be fired via autocmds.
+-- The table is set up such that:
+-- {
+--   'request/name' : {'BufWritePre', {'User', 'LSPRequestName'}}
+-- }
+--
+-- Also has a function to setup all the autocommands
+
+-- local requests = require('lsp.request').requests
+local log = require('lsp.log')
+local util = require('neovim.util')
+
+
+local autocmd_table = {
+  ['textDocument/didOpen'] = {
+    -- After initialization, make sure to tell the LSP that we opened the file
+    {'User', 'initialize/post'},
+
+    'BufReadPost',
+    -- 'BufEnter',
+  },
+
+  ['textDocument/willSave'] = {
+    'BufWritePre',
+  },
+
+  ['textDocument/didSave'] = {
+    'BufWritePost',
+  },
+
+  -- TODO(tjdevries): Not too familiar with close autocommands
+  ['textDocument/didClose'] = {
+    'BufDelete',
+    'BufWipeout',
+  },
+
+  ['textDocument/didChange'] = {
```

Comment:
I would be great if this was easily configurable. LangageClient-neovim showed me that doing this also on TextChangedI is a great idea (for a medium-size c file, but probably not for a c++ file full of template metaprogramming)